src/palimpzest/utils/data_helpers.py

```diff
@@ -0,0 +1,61 @@
+import os
+from pathlib import Path
+
+import pandas as pd
+
+from palimpzest import constants
+from palimpzest.core.data.datareaders import (
+    DataReader,
+    FileReader,
+    HTMLFileDirectoryReader,
+    ImageFileDirectoryReader,
+    MemoryReader,
+    PDFFileDirectoryReader,
+    TextFileDirectoryReader,
+    XLSFileDirectoryReader,
+)
+
+
+def get_local_source(path: str | Path, **kwargs) -> DataReader:
+    """Return a DataReader for a local file or directory."""
+    if os.path.isfile(path):
+        return FileReader(path)
+
+    elif os.path.isdir(path):
+        if all([f.endswith(tuple(constants.IMAGE_EXTENSIONS)) for f in os.listdir(path)]):
+            return ImageFileDirectoryReader(path)
+
+        elif all([f.endswith(tuple(constants.PDF_EXTENSIONS)) for f in os.listdir(path)]):
+            pdfprocessor = kwargs.get("pdfprocessor", constants.DEFAULT_PDF_PROCESSOR)
+            file_cache_dir = kwargs.get("file_cache_dir", "/tmp")
+            return PDFFileDirectoryReader(
+                path=path, pdfprocessor=pdfprocessor, file_cache_dir=file_cache_dir
+            )
+
+        elif all([f.endswith(tuple(constants.XLS_EXTENSIONS)) for f in os.listdir(path)]):
+            return XLSFileDirectoryReader(path)
+
+        elif all([f.endswith(tuple(constants.HTML_EXTENSIONS)) for f in os.listdir(path)]):
+            return HTMLFileDirectoryReader(path)
+
+        else:
```

Comment:
This is kinda bug-prone and could come back to bite us later.

I don't have a good solution though. Enforcing extensions for text files does not seem like a good idea.

Reply:
I agree -- if it's any consolation, the "`TextFileDirectoryReader`" calls `f.read()` on the each input file, so it should work on a pretty wide array of files.

But I agree that this approach is brittle and error prone -- I'd be happy to consider alternatives.